framework/web/UrlManager.php

```diff
@@ -126,11 +126,14 @@ class UrlManager extends Component
      */
     public $ruleConfig = ['class' => 'yii\web\UrlRule'];
 
+    /**
+     * @var array internal cache to store matched rules
+     */
+    protected $ruleCache;
```

Comment:
You are right, my example not helping to prove my opinion. But what if we want to count hits for specific rule. 
OR
we want to store only one rule, instead of array. 
 
```$this->ruleCache[$cacheKey] = $rule;```
instead of
```$this->ruleCache[$cacheKey][] = $rule;```

Well, maybe that cases are rare but can happen. I am not really against making it private but I think it can become sometimes a little pain for developers.